Sources/MapboxNavigation/InstructionPresenter.swift

```diff
@@ -98,13 +98,20 @@ class InstructionPresenter {
     
     func attributedTextRepresentation(of instruction: VisualInstruction,
                                       dataSource: DataSource,
-                                      imageRepository: ImageRepository,
-                                      onImageDownload: @escaping ImageDownloadCompletion) -> NSAttributedString {
+                                      spriteRepository: SpriteRepository,
+                                      onImageDownload: @escaping CompletionHandler) -> NSAttributedString {
         var components = instruction.components
         
         let isShield: (_ key: VisualInstruction.Component?) -> Bool = { (component) in
             guard let key = component?.cacheKey else { return false }
-            return imageRepository.cachedImageForKey(key) != nil
+            switch component {
+            case .image(let representation, _):
+                let image = spriteRepository.getShieldIcon(shield: representation.shield)
+                ?? spriteRepository.legacyCache.image(forKey: key)
```

Comment:
This case will need to prefer the legacy shield for state route shields for usability and for consistency with the current road name label: https://github.com/mapbox/mapbox-navigation-ios/pull/3723#discussion_r806328964.

Reply:
Yes, the `InstructionPresenter.attributedString(for representation:in:dataSource:onImageDownload:)` following this logic to generate the legacy shield preference for the  `us-state`. And this place here is to check whether the image is already stored or cached for the shield.